packages/core/src/new-api/internal/reconciliation/futures/reconcileArtifactContractAt.ts

```diff
@@ -0,0 +1,32 @@
+import { isEqual } from "lodash";
+
+import { ContractAtExecutionState } from "../../../types/execution-state";
+import { ArtifactContractAtFuture } from "../../../types/module";
+import { ExecutionStateResolver } from "../execution-state-resolver";
+import { ReconciliationContext, ReconciliationFutureResult } from "../types";
+import { fail } from "../utils";
+
+export function reconcileArtifactContractAt(
+  future: ArtifactContractAtFuture,
+  executionState: ContractAtExecutionState,
+  context: ReconciliationContext
+): ReconciliationFutureResult {
+  if (!isEqual(future.contractName, executionState.contractName)) {
+    return fail(
+      future,
+      `Contract name has been changed from ${executionState.contractName} to ${future.contractName}`
```

Comment:
Maybe these errors should include the id. In general, all of them.

You could have the same contract name in multiple modules.

Reply:
Oh, the future id is not lost.